src/components/ReadGallery/index.tsx

```diff
@@ -0,0 +1,94 @@
+import { useState } from 'react';
+import { getStorage, GalleryData } from 'apis/Gallery';
+import { useLocation } from 'react-router-dom';
+// import DeleteGallery from 'components/DeleteGallery';
+import styled from 'styled-components';
+import deleteIcon from '../../assets/icons/deleteIcon.png';
+
+function ReadGallery() {
+  const [galleryRead, setGalleryRead] = useState<GalleryData[]>([]);
+  getStorage(setGalleryRead);
```

Comment:
이 코드를 보면 아래와 같이 실행이 돼요!

getStorage 함수 실행 -> state 변경 -> state 변경으로 인한 리렌더링 -> 다시 getStorqage 함수 실행 -> state 변경 ...

이 과정이 반복되어서 무한 렌더링이 발생해요 ! 
렌더링이 될 때 변수 초기화가 되고 getStorage 같은 함수는 다시 실행하게 되어있어요 !! 프로젝트 끝나고 hook 공부해보시면 더 잘 아실 것 같아요 🙌🙌

저는 아래와 같은 코드로 무한렌더링을 해결했는데 수빈님 이미 코드 작성하신 것 같아서 그냥 참고하시면 좋을 것 같아요!!

```suggestion
   const getGalleryList = async () => {
    const result = await getStorage();
    if (result) setGalleryRead(result);
  };

  useEffect(() => {
    getGalleryList();
  }, []);
```

Reply:
참고로 저는 코드에서 result가 undefined로 들어올 상황에 대한 예외처리를 아예 해주지 않았어요!

Reply:
가현님.. 자세한 코드 설명 감사합니다.. 🙇🏻‍♀️👍
적용시켜보겠습니다!! 항상 많은걸 배워갑니다.. 👍👍